### PR TITLE
[enhancement](nereids)add eliminate filter on one row relation rule

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

1. simplify PushdownFilterThroughSetOperation rule
2. add eliminate filter on one row relation rule

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateFilter.java [hunk 1]
```diff
@@ -54,6 +64,35 @@ public Rule build() {
...
                 })
-                .toRule(RuleType.ELIMINATE_FILTER);
+                .toRule(RuleType.ELIMINATE_FILTER),
+                logicalFilter(logicalOneRowRelation()).thenApply(ctx -> {
+                    LogicalFilter<LogicalOneRowRelation> filter = ctx.root;
+                    Map<Expression, Expression> replaceMap =
+                            filter.child().getOutputs().stream().filter(e -> e instanceof Alias)
+                                    .collect(Collectors.toMap(NamedExpression::toSlot, e -> ((Alias) e).child()));
+
+                    Set<Expression> newConjuncts = Sets.newHashSet();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateFilter.java:74]: use ImmutableSet.Builder<Expression> to build conjuncts set

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by keanji-x — APPROVED** [review]

**Pull request merged**